jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/Utils.kt

```diff
@@ -285,3 +293,50 @@ fun isCodeModernizerAvailable(project: Project): Boolean {
 }
 
 fun isGradleProject(project: Project) = !GradleSettings.getInstance(project).linkedProjectsSettings.isEmpty()
+
+fun getJavaVersionFromProjectSetting(project: Project): String? = project.tryGetJdk()?.toString()
+
+fun getMavenVersions(project: Project): String {
+    fun getVersion(mavenCommand: String): String? {
+        try {
+            val commandLine = GeneralCommandLine(listOf(mavenCommand, "-v"))
+                .withWorkDirectory(project.basePath)
+                .withRedirectErrorStream(true)
+            val output = ExecUtil.execAndGetOutput(commandLine)
+            if (output.exitCode == 0) {
+                return parseMavenVersion(output.stdout)
+            } else {
+                CodeModernizerManager.LOG.error { "Failed to fetch $mavenCommand version: ${output.stdout}" }
+            }
+        } catch (e: ProcessNotCreatedException) {
+            CodeModernizerManager.LOG.warn { "$mavenCommand not set up" }
+        } catch (e: Exception) {
+            CodeModernizerManager.LOG.error(e) { "Failed to fetch $mavenCommand version" }
+        }
+        return null
+    }
+
+    // Get local maven version
+    val localMavenVersion: String? = getVersion("mvn")
+
+    // Get wrapper maven version
+    val mvnw = if (SystemInfo.isWindows) "./mvnw.cmd" else "./mvnw"
+    val wrapperMavenVersion: String? = getVersion(mvnw)
+
+    // Get user's Maven setting (using bundled vs local vs wrapper)
+    val mavenSettings = MavenProjectsManager.getInstance(project).getGeneralSettings()
+    val mavenHome = mavenSettings.getMavenHome()
+    // TODO: Need to detect bundled Maven version that come with IDEA
+    //  The utility returns "Use Maven wrapper" if using wrapper, "Bundled (Maven 3)" if using Bundled Maven, otherwise the local maven version.
+    val userMavenSetting = MavenUtil.getMavenVersion(mavenHome) ?: mavenHome
+
+    return "$wrapperMavenVersion (mvnw) -- $localMavenVersion (mvn) -- user setting: $userMavenSetting"
+}
+
+private fun parseMavenVersion(output: String?): String? {
+    if (output == null) return null
+    val mavenVersionIndex = output.indexOf("Apache Maven")
+    if (mavenVersionIndex == -1) return null
+    val mavenVersionString = output.slice(IntRange(mavenVersionIndex + 13, output.length - 1))
+    return mavenVersionString.slice(IntRange(0, output.indexOf(' ') - 1))
```

Comment:
```suggestion
    return try {
        val mavenVersionString = output.slice(IntRange(mavenVersionIndex + 13, output.length - 1))
    
        mavenVersionString.slice(IntRange(0, output.indexOf(' ') - 1))
    } catch (e: StringIndexOutOfBoundsException) {
        // also probably log an error here
        null
    }
```

Reply:
I think to add a Logger, I will need to introduce an object to wrap around all functions in the util file, then have to update imports across the code base, which will be a large change that doesn't seem suitable to add this PR, so I will leave the logging out.